src/main/java/com/cst438/service/RegistrarServiceProxy.java

```diff
@@ -0,0 +1,199 @@
+package com.cst438.service;
+
+import com.cst438.domain.*;
+import com.cst438.dto.CourseDTO;
+import com.cst438.dto.EnrollmentDTO;
+import com.cst438.dto.SectionDTO;
+import com.cst438.dto.UserDTO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Service
+public class RegistrarServiceProxy {
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private SectionRepository sectionRepository;
+
+    @Autowired
+    private TermRepository termRepository;
+
+    @Autowired
+    private GradeRepository gradeRepository;
+
+    @Autowired
+    private EnrollmentRepository enrollmentRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+
+    Queue registrarServiceQueue = new Queue("registrar_service", true);
+
+    @Bean
+    public Queue createQueue() {
+        return new Queue("gradebook_service", true);
+    }
+
+    @Autowired
+    RabbitTemplate rabbitTemplate;
+
+    public void updateFinalGrade(EnrollmentDTO e) {
+        sendMessage("updateFinalGrade " + asJsonString(e));
+    }
+
+    @RabbitListener(queues = "gradebook_service")
+    public void receiveFromRegistrar(String message)  {
+        try {
+            String[] parts = message.split(" ", 1);
```

Comment:
The split method should use limit 2 instead of 1 to correctly split the message into action and data parts.
Example:
String[] parts = message.split(" ", 2);

Reply:
Addressed this.